### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by mgrobertso**

### CODE REVIEW

File: crypto-api/Controllers/CryptoController.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+namespace crypto_api.Controllers
+{
+    [EnableCors("CorsPolicy")]
+    [ApiController]
+    [Route("api/[controller]")]
+
+    public class CryptoController : ControllerBase
+    {
+        private readonly DataContext _context;
```
Comment (swhook52) [inline crypto-api/Controllers/CryptoController.cs:16]: Same feedback here and probably other controllers... You want to move your datacontext to a service and simplify the controller. Right now all the business logic and data access and response logic is in one place. You need more separation of concerns which will allow for more testability when we get to unit testing.

**Review by swhook52 — CHANGES_REQUESTED** [review]

File: crypto-api/Controllers/AuthController.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            user.UserName = Signup.Username;
+            user.Email = Signup.Email;
+            user.FirstName = Signup.FirstName;
+            user.LastName = Signup.LastName;
+            user.PasswordHash = passwordHash;
+            user.PasswordSalt = passwordSalt;
+            await _context.Users.AddAsync(user);
+            await _context.SaveChangesAsync();
+            
+            return Ok("User has been Created");
```
[the remark this answers is not in the document]
Comment (mgrobertso) [reply crypto-api/Controllers/AuthController.cs:48]:
```
            return Ok(new RegisterResponse
            {
                Id = user.Id,
                Firstname = user.FirstName,
                Lastname = user.LastName,
                Username = user.UserName,
                Email = user.Email
            });
```

File: Crypto.Core/Repository/GenericRepository.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
+﻿using Crypto.Data;
```
Comment (irwinhurstCEI) [inline Crypto.Core/Repository/GenericRepository.cs:1]: Move any methods you need to your Service and delete this.

File: Crypto.Core/Repository/IUnitOfWork.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using Crypto.Core.DTOs;
```
Comment (irwinhurstCEI) [inline Crypto.Core/Repository/IUnitOfWork.cs:1]: DBContext is already a Unit Of Work so you dont need this abstraction,  Your Controllers should use your Services
https://learn.microsoft.com/en-us/dotnet/api/system.data.entity.dbcontext?view=entity-framework-6.2.0

File: crypto-api/Controllers/AuthController.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
+﻿using AutoMapper;
```
Comment (irwinhurstCEI) [inline crypto-api/Controllers/AuthController.cs:1]: REplace the UnitOfWork with a reference to your IAuthService

File: Crypto.Core/Services/AuthService.cs [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+        public User GetUserId(Guid id)
+        {
+            var user = _context.Users.SingleOrDefault(x => x.Id == id);
+
+            if (user == null)
+            {
+                throw new NotImplementedException();
+            }
+            return user;
```
Comment (irwinhurstCEI) [inline Crypto.Core/Services/AuthService.cs:75]: Your services should return DTO objects.

File: Crypto.Core/Services/ICryptoService.cs [hunk 7]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Crypto.Core.Services
```
Comment (irwinhurstCEI) [inline Crypto.Core/Services/ICryptoService.cs:1]: I think you are missing GetCryptoAsync() in your interface

**Review by swhook52 — COMMENTED** [review]
I see Irwin commented on some of the same things. just adding my notes as well.

**Review by swhook52 — APPROVED** [review]

**Pull request merged**